### PR TITLE
Refactor sending into emailable and add to payments

### PR DESCRIPTION
**Pull request by iHiD**
This allows objects other than notifications to have emails sent about them, by adding an `emailable` concern. The example use-case also implemented here is donation payments, but this is relevant to lots of things where it doesn't make sense to add a notification, but we do want to send an email.

### CODE REVIEW

File: test/models/user/notifications/donation_subscription_created_notification_test.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class User::Notifications::DonationSubscriptionCreatedNotificationTest < ActiveSupport::TestCase
+  include ActionView::Helpers::AssetUrlHelper
+  include Webpacker::Helper
+
+  test "keys are valid" do
+    user = create :user
+    payment = create :donation_payment, user: user
+
+    notification = User::Notifications::DonationSubscriptionCreatedNotification.create!(user: user, params: { payment: payment })
+    assert_equal "#{user.id}|acquired_badge|Badge##{badge.id}", notification.uniqueness_key
```
Comment (ErikSchierboom) [inline test/models/user/notifications/donation_subscription_created_notification_test.rb]: Where does `badge.id` come from?

File: db/migrate/20210908203439_add_email_to_donations.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddEmailToDonations < ActiveRecord::Migration[6.1]
+  def change
+    add_column :donations_payments, :email_status, :integer, limit: 1, default: 0, null: false
+    add_column :donations_subscriptions, :email_status, :integer, limit: 1, default: 0, null: false
```
Comment (ErikSchierboom) [inline db/migrate/20210908203439_add_email_to_donations.rb:4]: Will we retroactively send emails to people who donated?

File: app/models/user/notifications/donation_subscription_created_notification.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      def url
+        Exercism::Routes.settings_url
+      end
+
+      def image_type; end
+
+      def image_url; end
+
+      def guard_params
+        "Badge##{user_acquired_badge.badge_id}"
```
Comment (ErikSchierboom) [inline app/models/user/notifications/donation_subscription_created_notification.rb]: Why does the guard here use the badge ID? Is that for uniqueness?

**Review by ErikSchierboom — APPROVED** [review]

File: app/models/user/notifications/donation_subscription_created_notification.rb (hunk 3 above)
Comment (iHiD) [reply app/models/user/notifications/donation_subscription_created_notification.rb]: This file should have been deleted :+1:

File: db/migrate/20210908203439_add_email_to_donations.rb (hunk 2 above)
Comment (iHiD) [reply db/migrate/20210908203439_add_email_to_donations.rb:4]: Yeah - I'm going to do that once we push this out.

File: test/models/user/notifications/donation_subscription_created_notification_test.rb (hunk 1 above)
Comment (iHiD) [reply test/models/user/notifications/donation_subscription_created_notification_test.rb]: Should also have been deleted! :)

**Pull request merged**